src/GingerbreadCompetition/Pages/TeamSelector.jsx

```diff
@@ -0,0 +1,112 @@
+import { useState } from "react";
+import TeamPage from "./TeamsPage";
+import { names } from "../Data/Names";
+
+function shuffleArray(array) {
+
+    for (let i = array.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    }
+  
+    return array;
+  }
+
+function createTeams(names, teamCount) {
+    if (!Array.isArray(names) || names.length === 0) {
+        return (
+            <div>
+                <p>No names provided.</p>
+            </div>
+        );
+        
+    }
+    if (teamCount <= 0 || teamCount > names.length) {
+        console.log(teamCount)
+        console.log(names.length)
+        return (
+            <div>
+                <p>Invalid number of teams selected.</p>
+                <p>Please select a number between 1 .</p>
+            </div>
+        )
+        
+    }
+    const shuffledNames = shuffleArray([...names]);
+    const teams = Array.from({ length: teamCount }, () => []);
+    shuffledNames.forEach((name, index) => {
+        const teamIndex = index % teamCount;
+        teams[teamIndex].push(name);
+    })
+    return teams;
+  }
+
+
+function TeamSelector({ names }) {
+    const [teamCount, setTeamCount] = useState(0); // State to store selected number of teams
+    const [selected, setSelected] = useState(null);
+    const [availableNumbers, setAvailableNumbers] = useState([4, 5, 6]);
+
+    const handleTeamSelect = (e) => {
+        const value = Number(e.target.value);
+        if (!names || names.length < value) {
+            alert("Please add more names before creating teams. if attempting to teams of 6 you must have 6 or more names");
+            return null;
+        }
+        setTeamCount(value); // Update team count when a button is clicked
+        setSelected(value);
+        setAvailableNumbers([value]);
+    };
+    const resetSelction = () => {
+      setTeamCount(0);
+      setSelected(null);
+      setAvailableNumbers([4, 5, 6]);
+    };
+
+
+    
+    const renderTeams = () => {
+      if (teamCount === 0) return null; // No teams selected yet
+      const teams = createTeams(names, teamCount);
+      return (
+          <div className="team-grid">
+              {teams.map((team, index) => (
+                  <div key={index} className="team-card">
+                      <h2>Team {index + 1}</h2>
+                      <TeamPage names={team} />
+                  </div>
+              ))}
+        </div>
+      );
+    };
+  
+    return (
+        <div>
+        <h3>Teams of:</h3>
```

Comment:
Right now you are creating the number of teams specified here.
Rather than teams of a given size.

Like if I say "teams of 5", that's actually producing 5 teams, not teams of 5 people.